alicloud/resource_alicloud_gpdb_instance.go

```diff
@@ -0,0 +1,272 @@
+
+	return nil
+}
+
+func resourceAlicloudGpdbInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	gpdbService := GpdbService{client}
+
+	request, err := buildGpdbCreateRequest(d, meta)
+	if err != nil {
+		return WrapError(err)
+	}
+	raw, err := client.WithGpdbClient(func(client *gpdb.Client) (interface{}, error) {
+		return client.CreateDBInstance(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_gpdb_instance", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	response, _ := raw.(*gpdb.CreateDBInstanceResponse)
+	addDebug(request.GetActionName(), response)
+	d.SetId(response.DBInstanceId)
+	if err := gpdbService.WaitForGpdbInstance(d.Id(), Running, DefaultLongTimeout); err != nil {
+		return WrapError(err)
+	}
+
+	return resourceAlicloudGpdbInstanceUpdate(d, meta)
+}
+
+func resourceAlicloudGpdbInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	gpdbService := GpdbService{client}
+
+	// Begin Update
+	d.Partial(true)
+
+	// Update Instance Description
+	if d.HasChange("description") {
+		request := gpdb.CreateModifyDBInstanceDescriptionRequest()
+		request.DBInstanceId = d.Id()
+		request.DBInstanceDescription = d.Get("description").(string)
+		raw, err := client.WithGpdbClient(func(gpdbClient *gpdb.Client) (interface{}, error) {
+			return gpdbClient.ModifyDBInstanceDescription(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw)
+		d.SetPartial("description")
+	}
+
+	// Update Security Ips
+	if d.HasChange("security_ip_list") {
+		ipList := expandStringList(d.Get("security_ip_list").(*schema.Set).List())
+		ipStr := strings.Join(ipList[:], COMMA_SEPARATED)
+		// default disable connect from outside
+		if ipStr == "" {
+			ipStr = LOCAL_HOST_IP
+		}
+		if err := gpdbService.ModifyGpdbSecurityIps(d.Id(), ipStr); err != nil {
+			return WrapError(err)
+		}
+		d.SetPartial("security_ip_list")
+	}
+
+	// Finish Update
+	d.Partial(false)
+
+	return resourceAlicloudGpdbInstanceRead(d, meta)
+}
+
+func resourceAlicloudGpdbInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	gpdbService := GpdbService{client}
+
+	request := gpdb.CreateDeleteDBInstanceRequest()
+	request.DBInstanceId = d.Id()
+
+	err := resource.Retry(10*5*time.Minute, func() *resource.RetryError {
+		raw, err := client.WithGpdbClient(func(client *gpdb.Client) (interface{}, error) {
+			return client.DeleteDBInstance(request)
+		})
+
+		if err != nil {
+			if IsExceptedErrors(err, []string{InvalidGpdbInstanceStatus}) {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		addDebug(request.GetActionName(), raw)
+		return nil
+	})
+	if err != nil {
+		if gpdbService.NotFoundGpdbInstance(err) {
```

Comment:
There is no need to use `gpdbService.NotFoundGpdbInstance(err)` just `IsExceptedErrors(err, []string{InvalidGpdbInstanceIdNotFound, InvalidGpdbNameNotFound})` is ok.